v-next/hardhat/src/internal/builtin-plugins/solidity/build-system/compiler/solcjs-runner.ts

```diff
@@ -28,7 +26,8 @@ async function main() {
   const solcjsPath = process.argv[2];
   const solc = await getSolcJs(solcjsPath);
 
-  const output = solc.compile(input);
+  // eslint-disable-next-line @typescript-eslint/consistent-type-assertions -- the input read from the stdin should be a string
+  const output = solc.compile(input as string);
```

Comment:
Looking at the `readStream` implementation, it seems we are already returning a `string`. Could we type the return statement to avoid casting it here?